File: algorithms/070.Climbing_Stairs/Climbing_Stairs.py
<<<<<<< HEAD
# Time:  O(n)
# Space: O(1)

class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        prev, current = 0, 1
        for i in xrange(n):
            prev, current = current, prev + current, 
        return current

if __name__ == "__main__":
    result = Solution().climbStairs(2)
    print result
=======
# Time:  O(n)
# Space: O(1)

class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        prev, current = 0, 1
        for i in xrange(n):
            prev, current = current, prev + current, 
        return current

if __name__ == "__main__":
    result = Solution().climbStairs(2)
    print result
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74

File: algorithms/070.Climbing_Stairs/Climbing_Stairs.java
<<<<<<< HEAD
public class Solution {
    public int climbStairs(int n) {
        if (n <= 1) {
            return n;
        }
        int last = 1, lastlast = 1;
        int now = 0;
        for (int i = 2; i <= n; i++) {
            now = last + lastlast;
            lastlast = last;
            last = now;
        }
        return now;
    }
=======
public class Solution {
    public int climbStairs(int n) {
        if (n <= 1) {
            return n;
        }
        int last = 1, lastlast = 1;
        int now = 0;
        for (int i = 2; i <= n; i++) {
            now = last + lastlast;
            lastlast = last;
            last = now;
        }
        return now;
    }
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
}